cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/aws-orchestrator/AwsOrchestrator.service.tsx

```diff
@@ -68,23 +71,77 @@ export class AwsOrchestrator {
         });
     }
 
-    static async initNodeProfile(props: AwsOrchestratorProps) {
+    static initControlPlaneNodeType(props: AwsOrchestratorProps, nodeProfile: string) {
         const { awsDispatch, setErrorObject, errorObject } = props;
-        try {
-            const nodeTypeList = await AwsService.getAwsNodeTypes();
-            const nodeProfileList: { [key: string]: string } = {
-                [AWS_NODE_PROFILE_NAMES.SINGLE_NODE]: '',
-                [AWS_NODE_PROFILE_NAMES.HIGH_AVAILABILITY]: '',
-                [AWS_NODE_PROFILE_NAMES.PRODUCTION_READY]: '',
-            };
-            Object.keys(nodeProfileList).forEach((nodeProfile) => {
-                nodeProfileList[nodeProfile] = AwsDefaults.setDefaultNodeType(nodeTypeList, nodeProfile);
-            });
-            saveSingleResourceObject(awsDispatch, RESOURCE.ADD_RESOURCES, AWS_FIELDS.NODE_TYPE, nodeProfileList);
-            setErrorObject(removeErrorInfo(errorObject, AWS_FIELDS.NODE_TYPE));
-        } catch (e) {
-            clearPreviousResourceData(awsDispatch, RESOURCE.ADD_RESOURCES, AWS_FIELDS.NODE_TYPE);
-            setErrorObject(addErrorInfo(errorObject, e, AWS_FIELDS.NODE_TYPE));
+        DefaultOrchestrator.initResources<string>({
+            resourceName: AWS_FIELDS.NODE_TYPE,
+            dispatch: awsDispatch,
+            errorObject,
+            setErrorObject,
+            fetcher: () => AwsService.getAwsNodeTypes(),
+            fxnSelectDefault: (nodeTypes: string[]) => AwsDefaults.selectDefaultNodeType(nodeTypes, getDefaultNodeTypes(nodeProfile)),
```

Comment:
Just one nit here: by including this `fxnSelectDefault`, we will populate the FORM section with a value for the field `NODE_TYPE`. However, I don't think we're using that as a field any more. We have instead AVAILABILITY_ZONE_1_NODE_TYPE, AVAILABILITY_ZONE_2_NODE_TYPE, and AVAILABILITY_ZONE_3_NODE_TYPE.
If I'm understanding correctly, there will be no separate FIELD of NODE_TYPE, so I would suggest eliminating the `fxnSelectDefault` here (and also maybe creating an AWS_RESOURCES enum and make NODE_TYPE one of the values, to replace the NODE_TYPE in the AWS_FIELDS enum)

Reply:
Yes sir. This NODE_TYPE is used to store node type for a certain node profile (e.g. SINGLE_NODE) 
But AVAILABILITY_ZONE_1_NODE_TYPE, AVAILABILITY_ZONE_2_NODE_TYPE, and AVAILABILITY_ZONE_3_NODE_TYPE are the node type for avaialbility zones.